03week/towersOfHanoi.js

```diff
@@ -19,24 +29,66 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  if (startStack === 'a' && endStack === 'b') {
+    let mover = stacks.a.pop();
+    stacks.b.push(mover);
+  } else if (startStack === 'a' && endStack === 'c') {
+    let mover = stacks.a.pop();
+    stacks.c.push(mover);
+  } else if (startStack === 'b' && endStack === 'a') {
+    let mover = stacks.b.pop();
+    stacks.a.push(mover);
+  } else if (startStack === 'b' && endStack === 'c') {
+    let mover = stacks.b.pop();
+    stacks.c.push(mover);
+  } else if (startStack === 'c' && endStack === 'a') {
+    let mover = stacks.c.pop();
+    stacks.a.push(mover);
+  } else {
+    let mover = stacks.c.pop();
+    stacks.b.push(mover);
+  }
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(start, end) {
+  const startStack = stacks[start];
+  const endStack = stacks[end];
+  if (startStack.length === 0) {
+    return false;
+  }
+  if (endStack.length === 0) {
+    return true;
+  } else if (startStack[startStack.length - 1] > endStack[endStack.length - 1]) {
+    return false;
+  } else {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  if (stacks.b.length === 4 || stacks.c.length === 4) {
+    console.log('You win!')
+    return true;
+  } else {
+    return false;
+  }
+}
 
+function reset() {
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  movePiece(startStack, endStack);
```

Comment:
you should only move the piece if your function is legal, and you should only check for win if your function is legal